s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/utils/S3OffsetManagerEntry.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright 2024 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.s3.source.utils;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import io.aiven.kafka.connect.common.source.OffsetManager;
+
+import com.google.common.base.Objects;
+
+public final class S3OffsetManagerEntry implements OffsetManager.OffsetManagerEntry<S3OffsetManagerEntry> {
+
+    // package private statics for testing.
+    // TODO make this package private after values in S3SourceTask are no longer needed
+    public static final String BUCKET = "bucket";
+    public static final String OBJECT_KEY = "objectKey";
+    public static final String TOPIC = "topic";
+    public static final String PARTITION = "partition";
+    public static final String RECORD_COUNT = "recordCount";
+
+    /**
+     * THe list of Keys that may not be set via {@link #setProperty(String, Object)}.
+     */
+    static final List<String> RESTRICTED_KEYS = List.of(RECORD_COUNT);
+    /** The data map that stores all the values */
+    private final Map<String, Object> data;
+    /** THe record count for the data map. Extracted here because it is used/updated frequently during processing */
+    private long recordCount;
+
+    private final String bucket;
```

Comment:
In some places a mention of 'bucketName'. can we make it consistent, and same with topic.